resource/traversers/dfu_impl_update.cpp

```diff
@@ -477,7 +477,15 @@ int dfu_impl_t::update (vtx_t root, std::shared_ptr<match_writers_t> &writers,
     unsigned int needs = m_graph_db->metadata.v_rt_edges[dom].get_needs ();
     m_color.reset ();
 
-    rc = upd_dfv (root, writers, needs, x, jobmeta, true, dfu);
+    if ((rc = upd_dfv (root, writers, needs, x, jobmeta, true, dfu)) > 0) {
+         uint64_t starttime = jobmeta.at;
+         uint64_t endtime = jobmeta.at + jobmeta.duration;
+         if (writers->emit_tm (starttime, endtime) == -1) {
+             m_err_msg += __FUNCTION__;
+             m_err_msg += ": emit_tm returned -1.\n";
```

Comment:
Nit: as far as I can tell, this function only ever returns 0 as of now, but in the future, it could return any `int`.  Might be a little more resilient to check for < 0:
```suggestion
         if (writers->emit_tm (starttime, endtime) < 0) {
             m_err_msg += __FUNCTION__;
             m_err_msg += ": emit_tm returned <0.\n";
```

Reply:
Good point.

Reply:
Actually, this is a part of `emit` methods (we have `emit_vtx` and `emit_edg`) and the error checking is all done against -1.  To be consistent, it seems to make sense to keep this code as is for now and then later when we address #618, we can reevaluate and change all of the callosities if necessary, instead of making one off change?